internal/internal_task_handlers.go

```diff
@@ -786,6 +794,9 @@ processWorkflowLoop:
 			heartbeatLoop:
 				for {
 					if delayDuration <= 0 {
+						heartbeatTimer.Stop()
```

Comment:
not sure if it is possible to see this line execute > 1 times consecutively
so perhaps
```go
if heartbeatTimer != nil {
	heartbeatTimer.Stop()
	heartbeatTimer = nil
}
```

BTW, this pile of code needs rewrite (3 for loops)

Reply:
good catch. fixed